src/concat.c

```diff
@@ -0,0 +1,334 @@
+        }
+        if (slow5File_i->compress->method != pressMethod) {
+            WARNING("[Skip file]: %s has a different compression type. Use merge (instead of concat) to merge files.",
+                    slow5_files[i].c_str());
+            slow5_close(slow5File_i);
+            continue;
+        }
+        if(run_id_capture){
+            run_id_capture = 0;
+            char* temp = slow5_hdr_get("run_id", 0, slow5File_i->header);
+            if(!temp){
+                ERROR("No run_id information found in %s.", slow5_files[i].c_str());
+                return EXIT_FAILURE;
+            }
+            run_id = std::string(temp);
+
+        }else{
+            char* run_id_check = slow5_hdr_get("run_id", 0, slow5File_i->header);
+            if(!run_id_check){
+                ERROR("No run_id information found in %s.", slow5_files[i].c_str());
+                return EXIT_FAILURE;
+            }
+            if(strcmp(run_id_check,run_id.c_str())){
+                WARNING("[Skip file]: %s has a different run_id. Use merge (instead of concat) to merge files.",
+                        slow5_files[i].c_str());
+                slow5_close(slow5File_i);
+                continue;
+            }
+        }
+        if (slow5File_i->compress->method != pressMethod) {
+            WARNING("[Skip file]: %s has a different compression type. Use merge (instead of concat) to merge files.",
+                    slow5_files[i].c_str());
+            slow5_close(slow5File_i);
+            continue;
+        }
+
+        if(header_write){
+            header_write = 0;
+            khash_t(slow5_s2s) *rg = slow5_hdr_get_data(0, slow5File_i->header); // extract jth read_group related data from ith slow5file
+            int64_t new_read_group = slow5_hdr_add_rg_data(slow5File->header, rg); //assumption0
+            if(new_read_group != 0){ //sanity check
+                ERROR("New read group number is not equal to 0. something's wrong\n%s", "");
+                return EXIT_FAILURE;
+            }
+            //now write the header to the slow5File.
+            if(slow5_hdr_fwrite(slow5File->fp, slow5File->header, format_out, pressMethod) == -1){
+                ERROR("Could not write the header to %s\n", arg_fname_out);
+                return EXIT_FAILURE;
+            }
+        }
+        if(format_out==SLOW5_FORMAT_ASCII) {
+            // BUFSIZE default is 8192 bytes
+            //writing to reads to the output
+            // BUFSIZE of 1 means one chareter at time
+            // good values should fit to blocksize, like 1024 or 4096
+            // higher values reduce number of system calls
+            char buf[BUFSIZ];
+            size_t size;
+            while ((size = fread(buf, 1, BUFSIZ, slow5File_i->fp))) {
+                fwrite(buf, 1, size, slow5File->fp);
+            }
+            slow5_close(slow5File_i);
+        }else{
+//            DEBUG("%d",101);
+            int64_t pos = ftell(slow5File_i->fp);
```

Comment:
@hasindu2008 for your attention.

Reply:
could you explain what you are doing here?